From 0c1919b031d60f2498a5bf0f4de6d8f9177ff802 Mon Sep 17 00:00:00 2001
From: Meticulus <theonejohnnyd@gmail.com>
Date: Wed, 12 Mar 2014 18:15:08 -0500
Subject: [PATCH] Fix: "Ok Google"

-It seems thats the "source" device is defined incorrectly
for this device. So we just switch to the known mic source.

-This might be better fixed in somewhere else but this works
until/if a better fix is found.

Change-Id: Ia02fe0a766f16633fc4fc0f516ebf4f21e660eaa
---
 core/jni/android_media_AudioRecord.cpp | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/core/jni/android_media_AudioRecord.cpp b/core/jni/android_media_AudioRecord.cpp
index c9882c4..891432e 100644
--- a/core/jni/android_media_AudioRecord.cpp
+++ b/core/jni/android_media_AudioRecord.cpp
@@ -197,9 +197,19 @@ android_media_AudioRecord_setup(JNIEnv *env, jobject thiz, jobject weak_this,
                 // Java channel masks map directly to the native definition
         jint audioFormat, jint buffSizeInBytes, jintArray jSession)
 {
-    //ALOGV(">> Entering android_media_AudioRecord_setup");
-    //ALOGV("sampleRate=%d, audioFormat=%d, channel mask=%x, buffSizeInBytes=%d",
-    //     sampleRateInHertz, audioFormat, channelMask, buffSizeInBytes);
+	//ALOGI(">> Entering android_media_AudioRecord_setup");
+    //ALOGI("sampleRate=%d, audioFormat=%d, channel mask=%x, buffSizeInBytes=%d, source=%d",
+    //     sampleRateInHertz, audioFormat, channelMask, buffSizeInBytes, source);
+#ifdef STE_HARDWARE
+	ALOGI(">> Entering android_media_AudioRecord_setup");
+    ALOGI("sampleRate=%d, audioFormat=%d, channel mask=%x, buffSizeInBytes=%d, source=%d",
+         sampleRateInHertz, audioFormat, channelMask, buffSizeInBytes, source);
+    /* We don't have mic source 1999 so switch to regular mic source 1*/
+    if(source == 1999){
+		source = 1;
+		ALOGI("STE_HARDWARE source switched from 1999 to 1");
+	}
+#endif
 
     if (!audio_is_input_channel(channelMask)) {
         ALOGE("Error creating AudioRecord: channel mask %#x is not valid.", channelMask);
-- 
1.9.3

